R/zchunk_L241.fgas.R

```diff
@@ -33,81 +32,245 @@ module_emissions_L241.fgas_DISABLED <- function(command, ...) {
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      mutate(input.emissions = round(value, emissions.DIGITS_EMISSIONS)) %>%
+      select(-GCAM_region_ID, -value) ->
+      L241.hfc_all
+
+    # L241.pfc: F-gas emissions for technologies in all regions.
     #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
+    # Remove anything that's zero in all base years for any technology, because no future
+    # coefs are read in for any techs.
     #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
+    # Then round future gas emissions and format the data frame.
+    L142.pfc_R_S_T_Yh %>%
+      group_by(GCAM_region_ID, supplysector, subsector, stub.technology, Non.CO2) %>%
+      filter(sum(value) != 0, year %in% BASE_YEARS) %>%
+      mutate(input.emissions = round(value, emissions.DIGITS_EMISSIONS), year = as.numeric(year)) %>%
+      ungroup() %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      select(-GCAM_region_ID, -value) ->
+      L241.pfc_all
+
+
+    # F-gas emissions factors for future years
     #
+    # First, create a subset of the cooling emission factors from 2010. Eventually
+    # these values will be used to estimate future emission factors by scaling with
+    # USA emission factors.
+    L141.hfc_ef_R_cooling_Yh %>%
+      filter(year == max(emissions.HFC_MODEL_BASE_YEARS)) %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") ->
+      L141.hfc_ef_cooling_2010
+
+
+    # From the 2010 hfc cooling emission factors select USA emission factors, in
+    # subsequent steps the USA emission factors will be used to estimate future
+    # emission factors.
+    L141.hfc_ef_cooling_2010 %>%
+      filter(region == "USA") %>%
+      select(USA_factor = value, -region, year, Non.CO2, supplysector) ->
+      L141.hfc_ef_cooling_2010_USA
+
+
+    # Match USA cooling emissions factor from 2010 with the other 2010
+    # hfc cooling emission factors by sector, gas, and year.
+    #
+    # Scale the the USA factor emissions for HFC134a by dividing by three since it
+    # is less commonly used now in USA.
+    L141.hfc_ef_cooling_2010 %>%
+      left_join_error_no_match(L141.hfc_ef_cooling_2010_USA, by = c("supplysector", "Non.CO2", "year")) %>%
+      mutate(USA_factor = if_else(Non.CO2 == "HFC134a", USA_factor / 3, USA_factor)) ->
+      L241.hfc_cool_ef_2010_USfactor
+
+
+    # Format the data frame of 2010 regional emission factors and 2010 USA emission factors
+    # for the next step where future emission factors are calculated.
+    #
+    # Future emission factors are will not be calculated for regions with 2010 emission factors
```

Comment:
this comment confuses me.  it could be that I don't know enough about the topic or the comment could use some clarification.